OpenMP problem with static library - Windows

I am trying to move a project, compiled with VS 16.8.4 and MSVC, from Intel MKL to OpenBLAS.
I have been following the instructions on 

https://github.com/xianyi/OpenBLAS/wiki/How-to-use-OpenBLAS-in-Microsoft-Visual-Studio

and they worked quite ok, with the sole addition of -DMSVC_STATIC_CRT=ON as option, to avoid linking problem when /MT is employed.

The annoying fact is that, when I use the libs in my project, compiled via MSVC,  at linking time it requires to link against (beside the Fortran stuff)

ompstub.lib

and at run time it requires

ompstub.dll

This seems to modify the behaviour of my code, as now omp_get_max_threads() returns 1.
Any clue on why that happens?
This seems to happen with develop and v.0.3.13 alike.
Moreover I get this warning while linking:

3>LINK : warning LNK4098: defaultlib 'msvcrt' conflicts with use of other libs; use /NODEFAULTLIB:library
but it is mostly due to the fact that OpenBLAS is compiled with clang-cl, I guess,

The option I pass to cmake are:

cmake .. -G "Ninja" -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_C_COMPILER=clang-cl -DCMAKE_Fortran_COMPILER=flang -DBUILD_WITHOUT_LAPACK=no -DNOFORTRAN=0 -DDYNAMIC_ARCH=ON -DCMAKE_BUILD_TYPE=Release -DMSVC_STATIC_CRT=ON -DCPP_THREAD_SAFETY_GEMV=ON

Unfortunately it does not work, as it seems not able to find Fortran OPENMP:
```
-- Compiling a 64-bit binary.
-- Found OpenMP_C: -Xclang -fopenmp (found version "3.1")
CMake Error at C:/Users/Enzo/miniconda3/Library/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find OpenMP_Fortran (missing: OpenMP_Fortran_FLAGS
  OpenMP_Fortran_LIB_NAMES)
Call Stack (most recent call first):
  C:/Users/Enzo/miniconda3/Library/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:582 (_FPHSA_FAILURE_MESSAGE)
  C:/Users/Enzo/miniconda3/Library/share/cmake-3.19/Modules/FindOpenMP.cmake:529 (find_package_handle_standard_args)
  cmake/arch.cmake:37 (find_package)
  cmake/system.cmake:217 (include)
  CMakeLists.txt:70 (include)
```
  I will try to make it work with ICC and IFortran, as they are now freely available.
  Beside that, is that intended that a non-parallel build has a dependency on (stub) OpenMP?

> Does it work with CMake from visual studio installer?

It requires an update of the ninja builder (does not support fortran), but after that yields exactly the same results.

The fix suggested :
```
  -DOpenMP_Fortran_FLAGS=-fopenmp \
  -DOpenMP_Fortran_LIB_NAMES=gomp \
  -DOpenMP_gomp_LIBRARY=-lgomp \
```
fixes the configuration step for the USE_OPENMP=1, but the build fails. For instance I get:
```
FAILED: CMakeFiles/openblas.dir/lapack-netlib/SRC/slagv2.f.obj
C:\Users\Enzo\miniconda3\Library\bin\flang.exe  -I..\lapack-netlib\SRC -I..\lapack-netlib\LAPACKE\include -fopenmp -fopenmp -Mrecursive -Kieee      -Mrecursive -Kieee -o CMakeFiles\openblas.dir\lapack-netlib\SRC\slagv2.f.obj -c CMakeFiles\openblas.dir\lapack-netlib\SRC\slagv2.f-pp.f
flang.exe: error: Fortran frontend to LLVM command failed due to signal (use -v to see invocation)
clang version 5.0.0
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Users\Enzo\miniconda3\Library\bin
flang.exe: note: diagnostic msg: PLEASE submit a bug report to  and include the crash backtrace, preprocessed source, and associated run script.
flang.exe: note: diagnostic msg: Error generating preprocessed source(s) - no preprocessable inputs.
ninja: build stopped: subcommand failed.
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -30,10 +30,11 @@ environment:
     CONDA_INSTALL_LOCN: C:\\Miniconda36-x64
   matrix:
     - COMPILER: clang-cl
-      WITH_FORTRAN: yes
+      WITH_FORTRAN: ON
+      USE_OPENMP: ON
     - COMPILER: clang-cl
       DYNAMIC_ARCH: ON
-      WITH_FORTRAN: no
+      WITH_FORTRAN: OFF
     - COMPILER: cl
     - COMPILER: MinGW64-gcc-7.2.0-mingw
       DYNAMIC_ARCH: OFF
@@ -47,12 +48,7 @@ environment:
 install:
   - if [%COMPILER%]==[clang-cl] call %CONDA_INSTALL_LOCN%\Scripts\activate.bat
   - if [%COMPILER%]==[clang-cl] conda config --add channels conda-forge --force
-  - if [%COMPILER%]==[clang-cl] conda install --yes --quiet clangdev cmake
-
-  - if [%WITH_FORTRAN%]==[no] conda install --yes --quiet ninja
-  - if [%WITH_FORTRAN%]==[yes] conda install --yes --quiet -c isuruf kitware-ninja
-  - if [%WITH_FORTRAN%]==[yes] conda install --yes --quiet flang
-
+  - if [%COMPILER%]==[clang-cl] conda install --yes --quiet clangdev cmake ninja flang=11.0.1
   - if [%COMPILER%]==[clang-cl] call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat" x64
   - if [%COMPILER%]==[clang-cl] set "LIB=%CONDA_INSTALL_LOCN%\Library\lib;%LIB%"
   - if [%COMPILER%]==[clang-cl] set "CPATH=%CONDA_INSTALL_LOCN%\Library\include;%CPATH%"
@@ -68,8 +64,9 @@ before_build:
   - if [%COMPILER%]==[MinGW64-gcc-7.2.0-mingw] cmake -G "MinGW Makefiles" -DNOFORTRAN=1 ..
   - if [%COMPILER%]==[MinGW-gcc-6.3.0-32] cmake -G "MSYS Makefiles" -DNOFORTRAN=1 ..
   - if [%COMPILER%]==[MinGW-gcc-5.3.0] cmake -G "MSYS Makefiles" -DNOFORTRAN=1 ..
-  - if [%WITH_FORTRAN%]==[no] cmake -G "Ninja" -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_C_COMPILER=clang-cl -DMSVC_STATIC_CRT=ON ..
-  - if [%WITH_FORTRAN%]==[yes] cmake -G "Ninja" -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_C_COMPILER=clang-cl -DCMAKE_Fortran_COMPILER=flang -DBUILD_WITHOUT_LAPACK=no -DNOFORTRAN=0 ..
+  - if [%WITH_FORTRAN%]==[OFF] cmake -G "Ninja" -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_C_COMPILER=clang-cl -DMSVC_STATIC_CRT=ON ..
+  - if [%WITH_FORTRAN%]==[ON] cmake -G "Ninja" -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_C_COMPILER=clang-cl -DCMAKE_Fortran_COMPILER=flang -DBUILD_WITHOUT_LAPACK=no -DNOFORTRAN=0 ..
+  - if [%USE_OPENMP%]==[ON] cmake -DUSE_OPENMP=ON ..
   - if [%DYNAMIC_ARCH%]==[ON] cmake -DDYNAMIC_ARCH=ON -DDYNAMIC_LIST='CORE2;NEHALEM;SANDYBRIDGE;BULLDOZER;HASWELL' ..
 
 build_script:

